server/debugsession.go

```diff
@@ -0,0 +1,834 @@
+				AllThreadsContinued: true,
+			},
+		})
+
+		// NOTE: ContinuedEvent not expected:
+		//   Please note: a debug adapter is not expected to send this event in response
+		//   to a request that implies that execution continues, e.g. ‘launch’ or ‘continue’.
+		//   It is only necessary to send a ‘continued’ event if there was no previous request that implied this.
+
+	case *dap.EvaluateRequest:
+		// TODO: reply with error if ds.stop == nil
+
+		variableName := request.Arguments.Expression
+
+		activation := ds.debugger.CurrentActivation(ds.stop.Interpreter)
+
+		variable := activation.Find(variableName)
+		if variable != nil {
+			value := variable.GetValue()
+			ds.send(&dap.EvaluateResponse{
+				Response: newDAPSuccessResponse(request.GetRequest()),
+				Body: dap.EvaluateResponseBody{
+					Result: value.String(),
+				},
+			})
+		}
+
+		ds.send(newDAPErrorResponse(
+			request.Seq,
+			request.Command,
+			dap.ErrorMessage{
+				Format: "unknown variable: {name}",
+				Variables: map[string]string{
+					"name": variableName,
+				},
+			},
+		))
+
+	case *dap.ScopesRequest:
+		// TODO: return more fine-grained scopes
+
+		ds.send(&dap.ScopesResponse{
+			Response: newDAPSuccessResponse(request.GetRequest()),
+			Body: dap.ScopesResponseBody{
+				Scopes: []dap.Scope{
+					{
+						Name:               "Variables",
+						PresentationHint:   "locals",
+						VariablesReference: int(ScopeIdentifierLocal),
+					},
+					{
+						Name:               "Storage",
+						PresentationHint:   "registers",
+						VariablesReference: int(ScopeIdentifierStorage),
+					},
+				},
+			},
+		})
+
+	case *dap.VariablesRequest:
+		// TODO: reply with error if ds.stop == nil
+		if ds.stop == nil {
+			ds.send(newDAPErrorResponse(request.GetRequest().GetSeq(), "", dap.ErrorMessage{}))
+
+		}
+
+		variableRequested := request.Arguments.VariablesReference
+		responseVariables := make([]dap.Variable, 0)
+
+		inter := ds.stop.Interpreter
+
+		switch ScopeIdentifier(variableRequested) {
+
+		case ScopeIdentifierLocal:
+			//reset variables
+			ds.variableHandleCounter = 0
+			ds.variables = make(map[int]any, 0)
+
+			activation := ds.debugger.CurrentActivation(inter)
+			location := ds.stop.Interpreter.Location
+			functionValues := activation.FunctionValues()
+
+			for name, variable := range functionValues {
+
+				// TODO: generalize exclusion of built-ins / standard library definitions
+
+				if location.String() == ds.scriptID && name == "self" {
+					continue
+				}
+				if name == "BLS" || name == "RLP" {
+					continue
+				}
+
+				value := variable.GetValue()
+
+				cadenceValue, err := runtime.ExportValue(value, inter, interpreter.EmptyLocationRange)
+				if err != nil {
+					//	panic(err)
+					continue
+				}
+				variable := ds.convertValueToDapVariable(name, cadenceValue)
+				responseVariables = append(
+					responseVariables,
+					variable,
+				)
+			}
+
+		case ScopeIdentifierStorage:
+			var index int = 1
```

Comment:
```suggestion
			index := 1
```

Reply:
Refactored in 87c3211102178900f8925d4877b0a60f9ff986bb